apps/documentation/components/customComponents/CodeCopyButton.tsx

```diff
@@ -0,0 +1,41 @@
+'use client';
+
+import { Button } from '@codaco/ui';
+import { CopyCheck } from 'lucide-react';
+import { useState } from 'react';
+import { cn } from '~/lib/utils';
+
+const CodeCopyButton = ({
+  code,
+  className,
+}: {
+  code: string;
+  className?: string;
+}) => {
+  const [isCopied, setIsCopied] = useState(false);
+
+  const copyToClipboard = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text);
+      setIsCopied(true);
+      setTimeout(() => setIsCopied(false), 2000); // Reset state after 2 seconds
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('Failed to copy to clipboard:', error);
+    }
+  };
+
+  return (
+    <Button
+      size={'xs'}
+      variant={'tableHeader'}
+      className={cn('rounded-md bg-platinum p-2 text-navy-taupe', className)}
```

Comment:
This isn't a semantic use of this variant. Please only use variants for their intended use-case. This isn't a table header. If none of the existing variants work, you need to create another one. However in this case I don't see why this can't just be a regular xs button.

Same with classNames: you shouldn't be adding arbitrary classes to a button. This will create styling inconsistencies.

Reply:
I added the arbitrary classes based on your suggestion above: **"Perhaps give the button a light background? The text isn't readable right now."**

But it's okay because I removed the button variant and now using the default one. Thank you!